### ISSUE TITLE
IL2026, IL3050 show up in Error List, then disappear

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#79965]
**Version Used**: Visual Studio 2022 Update 14 (Dev17.14)

**Diagnostic Id**: IL2026, IL3050

> **IL3050** Using member 'StreamJsonRpc.JsonRpc.Attach<T>()' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. This code generates IL at runtime and executes it.
> **IL2026** Using member 'StreamJsonRpc.JsonRpc.Attach<T>()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. This code generates IL at runtime and executes it.

**Repro steps**:

```
git clone https://github.com/microsoft/vs-streamjsonrpc.git
cd vs-streamjsonrpc
git checkout 835ae62576a8d7f60c186a9a1a79dbc7e547857d
.\init.ps1
StreamJsonRpc.slnx
```

Open some code files that are _not_ in the `NativeAOTCompatibility.Test` project.

**Expected Behavior**:

No warnings appear in the error list, consistent with build/publish at the CLI.

**Actual Behavior**:

While working in StreamJsonRpc, I frequently see this:
 
<img width="2804" height="288" alt="Image" src="https://github.com/user-attachments/assets/164ae0dd-7aee-4065-b699-cc5c248d9c78" />

But when I double-click on them to see the source code, they vanish.
The linker does not produce any such warnings. So it seems these roslyn analyzers are defective.
I have "Entire Solution" set for both "Background code analysis" and "Show compiler errors and warnings for", BTW.

**Comment by agocke** [comment]
> So it seems these roslyn analyzers are defective.

Note: they _may_ be defective, but if they are, I don't know why, I don't know how to figure that out, and I'm not aware of any documentation on this topic. So I could use help from Roslyn identifying what's going on in this scenario that makes it different from build.

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/26817

**Comment by jjonescz** [comment]
If you can get a binlog of the design-time build where the warnings appear, that might be helpful.
